pythonx/UltiSnips/snippet/parsing/lexer.py

```diff
@@ -250,6 +250,41 @@ def __repr__(self):
         return "MirrorToken(%r,%r,%r)" % (self.start, self.end, self.number)
 
 
+class ChoicesToken(Token):
+
+    """${1|o1,o2,o3|}"""
```

Comment:
Worthwile adding in this comment that choices are taken literally, i.e. no recursive parsing is taking place like it is for other tabstops.